src/CommandContainer.js

```diff
@@ -2,13 +2,31 @@ const Class = require('./Class');
 
 module.exports = Class.extend({
     commandList: {},
-    add: function (key, command) {
+    aliasList: {},
+    addCommand: function (key, command) {
         this.commandList[key] = command;
     },
+    addAlias: function (key, commandName) {
+        this.aliasList[key] = commandName;
+    },
+    hasCommand: function (key) {
+        return !!this.commandList[key];
+    },
+    hasAlias: function (key) {
+        return !!this.aliasList[key];
+    },
     get: function (key) {
-        return this.commandList[key];
+        if (this.hasCommand(key)) {
+            return this.commandList[key];
+        } else if (this.hasAlias(key)) {
+            return this.commandList[this.aliasList[key]];
+        }
+        return null;
```

Comment:
I know you wanted to keep this simplified as a single list.  Why not have addCommand also add to the aliasList?  `alias` might be a misnomer at that point, but you'd only have one lookup here.

Reply:
Hm, I could do that, as long as the list of commands without the aliases are still available. I could also add everything to a single map and then just have an array of all commandNames, if that is better. Pretty much how it was earlier, but with an additional list that stores only the names of the commands and a map with the actual classes behind the names + aliases, if that is better.

Reply:
Another possibility is to add all commands + aliases to a single list, and have the !commands command filter out all duplicate entries when only looking at commandName.